tests/test-webapps/test-servlet-spec/test-spec-webapp/src/main/java/com/acme/test/TestListener.java

```diff
@@ -165,84 +168,66 @@ public void contextInitialized(ServletContextEvent sce)
     @Override
     public void contextDestroyed(ServletContextEvent sce)
     {
-        // System.err.println("contextDestroyed "+sce);
+        LOG.fine("contextDestroyed " + sce);
     }
 
     @Override
     public void attributeAdded(ServletContextAttributeEvent scab)
     {
-        // System.err.println("attributeAdded "+scab);
+        LOG.fine("attributeAdded " + scab);
     }
 
     @Override
     public void attributeRemoved(ServletContextAttributeEvent scab)
     {
-        // System.err.println("attributeRemoved "+scab);
+        LOG.fine("attributeRemoved " + scab);
     }
 
     @Override
     public void attributeReplaced(ServletContextAttributeEvent scab)
     {
-        // System.err.println("attributeReplaced "+scab);
+        LOG.fine("attributeReplaced " + scab);
     }
 
     @Override
     public void requestDestroyed(ServletRequestEvent sre)
     {
-        // System.err.println("requestDestroyed "+sre);
+        LOG.fine("requestDestroyed " + sre);
     }
 
     @Override
     public void requestInitialized(ServletRequestEvent sre)
     {
-        // System.err.println("requestInitialized "+sre);
+        LOG.fine("requestInitialized " + sre);
     }
 
     @Override
     public void attributeAdded(ServletRequestAttributeEvent srae)
     {
-        // System.err.println("attributeAdded "+srae);
+        LOG.fine("attributeAdded " + srae);
     }
 
     @Override
     public void attributeRemoved(ServletRequestAttributeEvent srae)
     {
-        // System.err.println("attributeRemoved "+srae);
+        LOG.fine("attributeRemoved " + srae);
     }
 
     @Override
     public void attributeReplaced(ServletRequestAttributeEvent srae)
     {
-        // System.err.println("attributeReplaced "+srae);
+        LOG.fine("attributeReplaced " + srae);
     }
 
     @Override
     public void sessionCreated(HttpSessionEvent se)
     {
-        // System.err.println("sessionCreated "+se);
+        LOG.fine("sessionCreated " + se);
     }
 
     @Override
     public void sessionDestroyed(HttpSessionEvent se)
     {
-        // System.err.println("sessionDestroyed "+se);
-    }
-
-    public void requestCompleted(ServletRequestEvent rre)
-    {
-        // TODO Auto-generated method stub
-
-    }
-
-    public void requestResumed(ServletRequestEvent rre)
-    {
-        // TODO Auto-generated method stub
-
-    }
-
-    public void requestSuspended(ServletRequestEvent rre)
-    {
-        // TODO Auto-generated method stub
-
```

Comment:
As before, add logging statements for these too.

Reply:
These three methods are not called, by anything.

``` java
public void requestCompleted(ServletRequestEvent rre)
public void requestResumed(ServletRequestEvent rre)
public void requestSuspended(ServletRequestEvent rre)
```

Nothing in Jetty calls them.
Nothing in the Servlet spec calls them.

* Servlet-Api -> https://github.com/eclipse-ee4j/servlet-api/search?q=requestSuspended&unscoped_q=requestSuspended
* Jetty -> https://github.com/eclipse/jetty.project/search?q=requestSuspended&unscoped_q=requestSuspended

The only listener that accepts `ServletRequestEvent` is the `javax.servlet.ServletRequestListener`
https://github.com/eclipse-ee4j/servlet-api/blob/4.0.4/api/src/main/java/javax/servlet/ServletRequestListener.java

Which doesn't have the 3 methods listed above.

Reply:
Apparently this came from an early beta level Servlet 3.0 spec.
For features being experimented with, but never became part of the spec.

See slide 28 -> https://www.oracle.com/technetwork/server-storage/ts-5415-159162.pdf

Reply:
OK, cool then.